app/api.go

```diff
@@ -55,6 +55,10 @@ func generateWeights(
 	topicId uint64) {
 	inferencesByTimestamp := []LatestInferences{}
 
+	//
+	// TODO generate new nonce and add to request!
+	//
+
```

Comment:
Also, we need to create an unfulfilledNonce to be actually inserted.

Reply:
I thought the losses were gonna be read by the reputers?

No matter though, you can use these functions to get the network loss at a given block or at or just before a given block:

```
GetNetworkLossBundleAtBlock
GetNetworkLossBundleAtOrBeforeBlock
```

Reply:
And that losses data looks almost right... the shape is the same but the property names and type of addresses are different than what we need in the chain...

Look for `ReputerValueBundles` in `types.proto` and look at its dependent protos, `ValueBunde`, etc.

Notably the cosmos address should be used instead of the libp2p key in the `node` field (property labelled `worker` too).